src/replica/replica_stub.cpp

```diff
@@ -2254,6 +2271,229 @@ void replica_stub::open_service()
     register_ctrl_command();
 }
 
+#if !defined(DSN_ENABLE_GPERF) && defined(DSN_USE_JEMALLOC)
+static std::string process_jemalloc_arena_decay_ms_command(const std::vector<std::string> &args,
+                                                           dsn::utils::je_decay_type decay_type)
+{
+    if (args.empty()) {
+        return std::string("ERR: invalid arguments");
+    }
+
+    bool all = false;
+    int32_t index = 0;
+    if (args[0] == "ALL") {
+        all = true;
+    } else if (!dsn::buf2int32(args[0], index)) {
+        return std::string("ERR: invalid arena index");
+    } else if (index < 0) {
+        return std::string("ERR: invalid arena index");
+    }
+
+    if (args.size() < 2) {
+        // show current value
+        std::string info;
+        if (all) {
+            dsn::utils::je_ctl_get_all_arenas_decay_ms_info(decay_type, &info);
+        } else {
+            dsn::utils::je_ctl_get_arena_decay_ms_info(
+                static_cast<unsigned>(index), decay_type, &info);
+        }
+        return info;
+    }
+
+    int64_t decay_ms = 0;
+    if (args[1] == "DEFAULT") {
+        // set to default value
+        decay_ms = 10000;
+    } else if (!dsn::buf2int64(args[1], decay_ms)) {
+        return std::string("ERR: invalid decay ms");
+    }
+
+    std::string msg;
+    if (all) {
+        dsn::utils::je_ctl_set_all_arenas_decay_ms(
+            decay_type, static_cast<ssize_t>(decay_ms), &msg);
+    } else {
+        dsn::utils::je_ctl_set_arena_decay_ms(
+            static_cast<unsigned>(index), decay_type, static_cast<ssize_t>(decay_ms), &msg);
+    }
+    return msg;
+}
+
+static std::string process_jemalloc_gc_arena_command(const std::vector<std::string> &args)
```

Comment:
Thanks for you pull request. And I think you can split it into serveral pull requests, because this patch is too big

Reply:
> Thanks for you pull request. And I think you can split it into serveral pull requests, because this patch is too big

Ok, I'll first commit build-related code (e.g. cmake files and build scripts), and the remote_command tools for jemalloc will be provided later.